lib/tbot/workloadidentity/workloadattest/systemd_linux.go

```diff
@@ -0,0 +1,81 @@
+//go:build linux
+
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package workloadattest
+
+import (
+	"context"
+	"log/slog"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-systemd/v22/dbus"
+	"github.com/gravitational/trace"
+
+	workloadidentityv1pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/workloadidentity/v1"
+)
+
+// SystemdAttestor attests the identity of a Systemd service.
+type SystemdAttestor struct {
+	cfg    SystemdAttestorConfig
+	log    *slog.Logger
+	dialer func(context.Context) (dbusConn, error)
+}
+
+func NewSystemdAttestor(cfg SystemdAttestorConfig, log *slog.Logger) *SystemdAttestor {
```

Comment:
godoc.

should we move the log into `SystemdAttestorConfig`?
In most of our usecases, we actually assign the default logger if the logger is nill.

Reply:
Good catch! I've added the missing godoc comment, but I'm tempted to keep the logger as a separate argument for now because `cfg` contains user-supplied configuration rather than runtime dependencies.